### PR TITLE
Fix empty intersection in Plane. projection for extreme normal vectors

### PR DESCRIPTION
**Pull request by adarsh-priydarshi-5646**
#### References to other Issues or PRs

motivated by #28612

#### Brief description of what is fixed or changed

Fixed `IndexError` in `Plane.projection()` when normal vector has zero components. Replaced intersection-based approach with direct projection formula.

#### Other comments

Added test cases for edge cases.

#### Release Notes

* geometry
  * Fixed `IndexError` in `Plane.projection()` for planes with zero normal vector components.

### CODE REVIEW

**Comment by adarsh-priydarshi-5646** [comment]
please review.

File: sympy/geometry/tests/test_plane.py [hunk 1]
```diff
@@ -255,6 +255,12 @@ def test_dimension_normalization():
     assert Plane((1, 2, 1), (2, 1, 0), (3, 1, 2)
         ).intersection((2, 1)) == [Point(2, 1, 0)]
 
+    p = Plane(Point3D(0, 0, 0), normal_vector=(1, 10**10, 1))
```
Comment (smichr) [inline sympy/geometry/tests/test_plane.py]:
```suggestion
    p = Plane(Point3D(0, 0, 0), normal_vector=(10**-7, 0, 1))
```

File: sympy/geometry/tests/test_plane.py [hunk 2]
```diff
@@ -255,6 +255,12 @@ def test_dimension_normalization():
...
+    pt = Point3D(1, 1, 1)
```
Comment (smichr) [inline sympy/geometry/tests/test_plane.py]:
```suggestion
    pt = Point3D(1, 0, 0)
```
